test/src/main/java/org/apache/accumulo/test/shell/ShellServerIT.java

```diff
@@ -520,10 +520,12 @@ public void setIterOptionPrompt() throws Exception {
 
   protected void checkTableForProperty(final AccumuloClient client, final String tableName,
       final String expectedKey, final String expectedValue) throws Exception {
-    assertTrue(
-        Wait.waitFor(() -> client.tableOperations().getConfiguration(tableName).get(expectedKey)
-            .equals(expectedValue), 5000, 500),
-        "Failed to find expected value for key: " + expectedKey);
+    try {
```

Comment:
This seems like a step backwards in terms of usage. The original issue commented on making it easier to use the Wait api but having to now use a try/catch makes this pretty ugly. Maybe instead of replacing the behavior that returns false we could have an additional method (or flag) that supports throwing the exception so we support both ways depending on what the unit test writer wants to do.

Reply:
This one place had an exception message that I thought would be nice to keep - all other occurrences where just failing when they used assertTrue and there were some that did not test the return condition.

Keeping the exception message in this single case if its not adding much.

Reply:
Maybe we could provide an option to provide an error message for the IllegalStateException then? I feel like this is going to come up again where we want to have a custom error message that is thrown. So we should just allow providing the message for the IllegalStateException as an option with an overloaded method.

Reply:
Could just add a method that is something like the following and the existing method can just have a default.

```Java
public static void waitFor(final Condition condition, final long duration, final long sleepMillis, String errorMsg)
      throws Exception {
//pass error message to new IllegalStateException()
}
```

Reply:
Added in 0031099059